### PR TITLE
Remove best fit

### PR DESCRIPTION
**Pull request by djwatson**
This option saves a few CPU cycles, but potentially adds a lot of fragmentation - so much so that there are workarounds like max_active.  Instead, let's just drop it entirely.  It only made a difference in one service I tested (.3% cpu regression), while many services saw a memory win (also small, less than 1% mem P99)

### CODE REVIEW

**Comment by davidtgoldblatt** [comment]
Love it. That being said, maybe let's wait until after we cut the release to merge, in case we want to get any last-minute fixes in cleanly?

**Comment by interwq** [comment]
Yeah let’s wait until we cut 5.2. I’m still curious about the effects of this on long running binaries.

**Comment by djwatson** [comment]
kept the max_active_fit check in first_fit path instead of removing.

File: src/extent.c [hunk 1]
```diff
@@ -483,7 +459,16 @@ extents_first_fit_locked(tsdn_t *tsdn, arena_t *arena, extents_t *extents,
 		assert(!extent_heap_empty(&extents->heaps[i]));
 		extent_t *extent = extent_heap_first(&extents->heaps[i]);
 		assert(extent_size_get(extent) >= size);
-		if (ret == NULL || extent_snad_comp(extent, ret) < 0) {
+                bool size_ok = true;
+		/*
+		 * In order to reduce fragmentation, avoid reusing and splitting
+		 * large extents for much smaller sizes.
+		 */
+		if ((sz_pind2sz(i) >> opt_lg_extent_max_active_fit) > size) {
```
Comment (interwq) [inline src/extent.c]: The intention was to only apply this to dirty_extents, maybe add a new arg to `extents_fit_locked` and specify if max_active_fit should be checked?

Comment (djwatson) [reply src/extent.c]: Sure, update with restriction to delay_coalesce (which is only set for dirty).

**Comment by interwq** [comment]
Should we land this and then #1455 next?

**Pull request closed without merging**